test/FastTests/Voron/Tables/DynamicBTreeIndex.cs

```diff
@@ -0,0 +1,213 @@
+                SetHelper(docs, "users/1", "Users", 1L, "{'Name': 'Oren'}");
+
+                tx.Commit();
+            }
+
+            using (var tx = Env.WriteTransaction())
+            {
+                var docs = tx.OpenTable(DocsSchema, "docs");
+
+                Slice.From(tx.Allocator, "users/1", out Slice key);
+                docs.DeleteByKey(key);
+
+                tx.Commit();
+            }
+
+            using (var tx = Env.ReadTransaction())
+            {
+                var docs = tx.OpenTable(DocsSchema, "docs");
+
+                var reader = docs.SeekForwardFrom(DocsSchema.Indexes[IndexName], Slices.BeforeAllKeys, 0);
+                Assert.Empty(reader);
+            }
+        }
+
+        [Fact]
+        public void CanInsertThenUpdateByCustom()
+        {
+            using (var tx = Env.WriteTransaction())
+            {
+                DocsSchema.Create(tx, "docs", 16);
+
+                tx.Commit();
+            }
+
+            const string id = "users/1";
+            using (var tx = Env.WriteTransaction())
+            {
+                var docs = tx.OpenTable(DocsSchema, "docs");
+                SetHelper(docs, id, "Users", 1L, "{'Name': 'Oren'}");
+
+                tx.Commit();
+            }
+
+            using (var tx = Env.WriteTransaction())
+            {
+                var docs = tx.OpenTable(DocsSchema, "docs");
+
+                SetHelper(docs, id, "Users", 2L, "{'Name': 'Eini'}");
+
+                tx.Commit();
+            }
+
+            using (var tx = Env.ReadTransaction())
+            {
+                var docs = tx.OpenTable(DocsSchema, "docs");
+
+                bool gotValues = false;
+                foreach (var reader in docs.SeekForwardFrom(DocsSchema.Indexes[IndexName], Slices.BeforeAllKeys, 0))
+                {
+                    var key = reader.Key;
+
+                    using (Allocator.Allocate(sizeof(long) * 2, out var buffer))
+                    using (Slice.From(Allocator, id, out var idSlice))
+                    using (Slice.From(Allocator, EndianBitConverter.Big.GetBytes(200L), out var expectedEtag))
+                    {
+                        expectedEtag.CopyTo(buffer.Ptr);
+
+                        var expectedHash = Hashing.XXHash64.Calculate(idSlice.Content.Ptr, (ulong)(idSlice.Size));
+                        *(long*)(buffer.Ptr + sizeof(long)) = (long)expectedHash;
+
+                        var expected = new Slice(buffer);
+
+                        Assert.True(SliceComparer.AreEqual(expected, key));
+                    }
+
+                    var handle = reader.Result;
+                    Assert.Equal("{'Name': 'Eini'}", Encoding.UTF8.GetString(handle.Reader.Read(3, out int size), size));
+                    tx.Commit();
+                    gotValues = true;
+                    break;
+                }
+
+                Assert.True(gotValues);
+            }
+        }
```

Comment:
Can we have some tests which deal with bigger number of items? let's say 10_000 records

Reply:
done